test/pythonml/test_context.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import os
+import re
+import unittest
+import subprocess
+from unittest import mock
+
+
+class TestContext(unittest.TestCase):
+    """
+    Unit tests for the methods in the Context class defined in the settings.py module
+    """
+
+    def setUp(self):
+        with open('fixtures/settings.py', 'r') as file:
+            settings_template = file.readlines()
+        with open('settings.py', 'w') as file:
+            for line in settings_template:
+                line = re.sub("PROBLEM_CATEGORY_HERE", 'regression', line)
+                file.write(line)
+        import settings
+        self.context = settings.Context()
+
+    def tearDown(self):
+        for data in ['.job_context', 'settings.py']:
+            subprocess.call('rm -rf '+data, shell=True)
```

Comment:
Unless we need actually need to spawn a new shell process, try to avoid using the shell `shell=True`

Could do `subprocess.call(['rm', '-rf', data])` instead

Also happens in the `unittest.baseclass` definition